application/src/main/java/md/maib/retail/application/CampaignMapper.java

```diff
@@ -0,0 +1,29 @@
+package md.maib.retail.application;
+
+import md.maib.retail.model.campaign.*;
+import md.maib.retail.model.conditions.Rule;
+import org.threeten.extra.Interval;
+
+import java.util.Collection;
+
+public class CampaignMapper {
+    public static CampaignSomeInfo mapToCampaignSomeInfo(Campaign campaign) {
```

Comment:
Each result / dto class can contain static factory for construct it from Campaign (named as of or valueOf)
Also we can use simple constructor